src/lib/y2network/interface_config_builder.rb

```diff
@@ -0,0 +1,160 @@
+# Copyright (c) [2019] SUSE LLC
+#
+# All Rights Reserved.
+#
+# This program is free software; you can redistribute it and/or modify it
+# under the terms of version 2 of the GNU General Public License as published
+# by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+# FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License for
+# more details.
+#
+# You should have received a copy of the GNU General Public License along
+# with this program; if not, contact SUSE LLC.
+#
+# To contact SUSE LLC about this file by physical or electronic mail, you may
+# find current contact information at www.suse.com.
+require "yast"
+require "y2network/interface_defaults"
+
+module Y2Network
+  # Stores what's needed when creating a new configuration for an interface
+  class InterfaceConfigBuilder
+    # @return [String] Device name (eth0, wlan0, etc.)
+    attr_accessor :name
+    # @return [String] type which is intended to be build
+    attr_accessor :type
+
+    include Y2Network::InterfaceDefaults
+
+    # Constructor
+    #
+    # Load with reasonable defaults;
+    # see LanItems::new_item_default_options, LanItems::@SysconfigDefaults and
+    # others as in LanItems::Select
+    def initialize
+      @config = init_device_config({})
+      @s390_config = init_device_s390_config({})
+    end
+
+    # Stores one option value tuple
+    # FIXME: obsolete by []=
+    def set(option: nil, value: nil)
+      @config[option] = value
+    end
+
+    # Returns currently stored option value
+    # FIXME: obsolete by []
+    def option(option)
+      @config.fetch(option, nil)
+    end
+
+    def []=(key, value)
```

Comment:
BTW this should be also in long term obsolete as it is very unreliable interface just acting like hash. But good compatible starting point.

Reply:
yes, but for now it made integration of your widgets much easier.